src/widgets/index/index.ts

```diff
@@ -402,11 +410,19 @@ const index = (props: IndexProps): Index => {
       // widgets. When the subscription happens before the `init` step, the (static)
       // configuration of the widget is pushed in the URL. That's what we want to avoid.
       // https://github.com/algolia/instantsearch.js/pull/994/commits/4a672ae3fd78809e213de0368549ef12e9dc9454
-      helper.on('change', ({ state }) => {
-        localUiState = getLocalWidgetsState(localWidgets, {
-          searchParameters: state,
-          helper: helper!,
-        });
+      helper.on('change', ({ state, isPageReset }) => {
+        if (isPageReset) {
+          localUiState.page = undefined;
```

Comment:
Could this be passed in the `getLocalWidgetsState` 3rd argument call below? That would be one less mutation and code branch to think about.

Reply:
how would you move it to the lower argument? I could only see this as an alternative, but I like it less:


```
const { page, ...stateWithoutPage } = localUiState;

// ...
  isPageReset ? stateWithoutPage : localUiState
```

Reply:
or else thought of putting it in here: https://github.com/algolia/instantsearch.js/blob/9ff71e2341acc5b09133e095c3285a2de162b846/src/widgets/index/index.ts#L364-L368